### PR TITLE
Migrate RemoteRuntimeDataset to the new SPI

### PR DESCRIPTION
**Pull request by jayfjin**
Migrate RemoteRuntimeDataset to the new SPI. Also wait until the JDBC driver is done to create the SQL unit test, since at that point we can simply pass in a different cconf. 

Build is green: https://builds.cask.co/browse/CDAP-DUT6842-1

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java [hunk 1]
```diff
@@ -396,8 +388,8 @@ private void clearStates() {
     }
     // TODO: CDAP-14848 merge transactions back together
```
Comment (poornachandra) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java]: Can the transactions be merged together now?

**Review by poornachandra — APPROVED** [review]
@jayfjin Just one comment, please fix and merge

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java (hunk 1 above)
Comment (jayfjin) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java]: Ah yes, thanks!

**Comment by jayfjin** [comment]
ITN is green https://builds.cask.co/browse/IT-ITM-105

**Pull request merged**